pkg/v1/stream/layer.go

```diff
@@ -159,19 +157,39 @@ func newCompressedReader(l *Layer) (*compressedReader, error) {
 	}
 
 	cr := &compressedReader{
-		closer: newMultiCloser(zw, l.blob),
-		pr:     pr,
-		bw:     bw,
-		h:      h,
-		zh:     zh,
-		count:  count,
-		l:      l,
+		pr:    pr,
+		bw:    bw,
+		h:     h,
+		zh:    zh,
+		count: count,
+		l:     l,
 	}
 	go func() {
-		if _, err := io.Copy(io.MultiWriter(h, zw), l.blob); err != nil {
-			pw.CloseWithError(err)
+		_, copyErr := io.Copy(io.MultiWriter(h, zw), l.blob)
+
+		// TODO: Why does flushing zw before closing cause TestStreamVsBuffer
+		// to fail? Is the test correct? Has the behavior of gzip Writer
+		// changed? The docs claim that Close does a flush, but that doesn't
+		// look like the case in 1.17.2.
+		//
+		// flushErr := zw.Flush()
```

Comment:
@jonjohnsonjr Do you have any idea what's going on here? Flushing before closing shouldn't cause a test to fail.

Reply:
I'm not sure what to make of this: https://go.dev/play/p/LNvLhKpO09n

Reply:
Ah, it looks like the additional 5 bytes code for... nothing. And they're before the footer. So calling flush before close in this case is wasteful. I'll remove this comment.